### ISSUE TITLE
how do I specify authentication

### ISSUE THREAD

**Issue by rwsdwatson** [git-for-windows/get-azure-pipelines-artifact#768]
I don't see how this action authenticates against the Azdo project.
If i just add it to a pipeline it fails with :-

`Error: Got code 203, URL: https://dev.azure.com/git/git/_apis/build/builds?definitions=1258&statusFilter=completed&resultFilter=succeeded&reasonFilter=all&$top=1, message: Non-Authoritative Information
`

**Comment by rimrul** [comment]
> I don't see how this action authenticates 

I don't think it does. But we do have some [azure piplines authentication code elsewhere](https://github.com/git-for-windows/gfw-helper-github-app/blob/main/GitForWindowsHelper/azure-pipelines.js) that we could probably port over here.

**Comment by dscho** [comment]
> Got code 203

That does not exactly say that authentication is required...

> > I don't see how this action authenticates
> 
> I don't think it does.

Indeed. This Action currently only supports public Azure Pipelines.

> But we do have some [azure piplines authentication code elsewhere](https://github.com/git-for-windows/gfw-helper-github-app/blob/main/GitForWindowsHelper/azure-pipelines.js) that we could probably port over here.

I suspect that it might be a bit more complicated than that, as it seems that Azure Pipelines is moving away from Personal Access Tokens (indeed, [the documentation currently only mentions OAuth 2.0](https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/list?view=azure-devops-rest-7.0#security) as way to authenticate).

Maybe we can integrate with https://github.com/azure/login instead of rolling our own authentication here? Then you should be able to use Managed Identities, which are quite safe from a security perspective because you can reduce the scope quite stringently.

**Comment by dscho** [comment]
> https://dev.azure.com/git/git/_apis/build/builds?definitions=1258&statusFilter=completed&resultFilter=succeeded&reasonFilter=all&$top=1

Are you sure that you specified the correct Azure DevOps project/repository? I ask because the `git/git` project has no [Azure Pipeline with ID 1258](https://dev.azure.com/git/git/_build?definitionId=1258):

![Image](https://github.com/user-attachments/assets/9236c4be-4570-47b6-9772-db411db496e2)

**Comment by rimrul** [comment]
> > https://dev.azure.com/git/git/_apis/build/builds?definitions=1258&statusFilter=completed&resultFilter=succeeded&reasonFilter=all&$top=1
> 
> Are you sure that you specified the correct Azure DevOps project/repository? I ask because the `git/git` project has no [Azure Pipeline with ID 1258](https://dev.azure.com/git/git/_build?definitionId=1258):
> 
> ![Image](https://github.com/user-attachments/assets/9236c4be-4570-47b6-9772-db411db496e2)

That seems to be an artefact from annoymization.

**Comment by rimrul** [comment]
> Maybe we can integrate with https://github.com/azure/login instead of rolling our own authentication here? 

That's only a wrapper around Azure CLI and Azure Powershell. I think neither of those leaves a usable session for us, but there is the [Azure Identity client library for JavaScript](https://learn.microsoft.com/en-us/javascript/api/overview/azure/identity-readme?view=azure-node-latest)

**Comment by rwsdwatson** [comment]
> > https://dev.azure.com/git/git/_apis/build/builds?definitions=1258&statusFilter=completed&resultFilter=succeeded&reasonFilter=all&$top=1
> 
> Are you sure that you specified the correct Azure DevOps project/repository? I ask because the `git/git` project has no [Azure Pipeline with ID 1258](https://dev.azure.com/git/git/_build?definitionId=1258):
> 
> ![Image](https://private-user-images.githubusercontent.com/127790/362801763-9236c4be-4570-47b6-9772-db411db496e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMTU5ODAsIm5iZiI6MTcyNTAxNTY4MCwicGF0aCI6Ii8xMjc3OTAvMzYyODAxNzYzLTkyMzZjNGJlLTQ1NzAtNDdiNi05NzcyLWRiNDExZGI0OTZlMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMFQxMTAxMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjFlYTM4M2UwZDcxYzliNGNiMzliYWE2ZWEzNmE1NmJhZmQxN2NlMDRkZTQ0NThkMDg2NDdiYjU5OGFiYTU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.zd5A696-QhHukxjzGiACSw29w9ly3NINjuIcKo6svAs)

Sorry I renamed my org / project.
